framework/platform/bpl/uci/scripts/platform_daemon_ugw.sh

```diff
@@ -0,0 +1,146 @@
+#!/bin/sh
+ap_to_repeater() 
+{
+    #configure APM
+    #uci set beerocks.config.management_mode="Proprietary-Mesh"
+    ubus call uci set '{ "config" : "beerocks" ,"section" : "config", "values" : {"management_mode" : "Proprietary-Mesh"}}'
+    #uci set beerocks.config.operating_mode="WDS-Repeater"
+    ubus call uci set '{ "config" : "beerocks" ,"section" : "config", "values" : {"operating_mode" : "WDS-Repeater"}}'
+    #uci set beerocks.config.ssid='adam_wave_43'
+    ubus call uci set '{ "config" : "beerocks" ,"section" : "config", "values" : {"ssid" : "adam_GW_2.4G"}}'
+    #uci set beerocks.config.mode_enabled='None'
+    ubus call uci set '{ "config" : "beerocks" ,"section" : "config", "values" : {"mode_enabled" : "None"}}'
+    #uci commit beerocks
+    ubus call uci commit '{"config" : "beerocks" }'
+    
+    static_init
+    
+    #configure wireless
+    # FH configuration for client connection (optional)
+    ubus call uci set '{ "config" : "wireless" ,"section" : "default_radio13", "values" : {"ssid" : "ssid_RPT_2.4G"}}'
+    ubus call uci set '{ "config" : "wireless" ,"section" : "default_radio13", "values" : {"key_mgmt" : "NONE"}}'
+    #2.4G
+    # 4addr-mode
+    ubus call uci set '{ "config" : "wireless" ,"section" : "default_radio26", "values" : {"wds" : "1"}}'
+    ubus call uci set '{ "config" : "wireless", "section" : "default_radio26", "values" : {"vendor_elems" : "13 dd050017353001 14 dd050017353001" } }'
+    # to remove above entry (use 3addr-mode) use: uci delete wireless.default_radio26.vendor_elems 
+    ubus call uci set '{ "config" : "wireless", "section" : "default_radio26", "values" : {"full_ch_master_control" : "1"}}'
+    ubus call uci set '{ "config" : "wireless" ,"section" : "default_radio26", "values" : {"disabled" : "0"}}'
+    
+    #5G
+    # 4addr-mode
+    ubus call uci set '{ "config" : "wireless" ,"section" : "default_radio58", "values" : {"wds" : "1"}}'
+    ubus call uci set '{ "config" : "wireless", "section" : "default_radio58", "values" : {"vendor_elems" : "13 dd050017353001 14 dd050017353001" } }'
+    
+    ubus call uci set '{ "config" : "wireless", "section" : "default_radio58", "values" : {"full_ch_master_control" : "1"}}'
+    ubus call uci set '{ "config" : "wireless" ,"section" : "default_radio58", "values" : {"disabled" : "0"}}'
+    
+    #uci commit wireless (currently automatically restarts the hostapd's)
+    ubus call uci commit '{"config" : "wireless" }'
+    # If the automatic restard is removed, the follwoing should be used after commit
+    #/etc/init.d/network reload
+}
+repeater_to_ap()
+{
+    # Kill APM
+    /opt/prplMesh/scripts/prplmesh_utils.sh stop
```

Comment:
It is NOT APM. It is prplMesh. Also, no need to comment at all, its pretty clear what is done here.
Also, do not use absolute path in the script, the installation path might change. Better to call `/etc/init.d/prplmesh.sh` to perform the repeater to AP settings, stop and start prplmesh.

So:

```
/etc/init.d/prplmesh.sh repeater_mode
/etc/init.d/prplmesh.sh restart
```

All the rest should be deleted.

Reply:
so you mean to move the prpl (originally beerocks) uci configurations to the prplmesh.sh script?

Reply:
yes, this is the platform script and is not part of prplmesh repository.
@arnout what do you think?

Reply:
I was the first one to comment that this script is a platform script and shouldn't be part of the prplmesh repository :-) Note that even openwrt-specific packages (ubus, uhttpd, ..) don't have their openwrt integration files as part of the package repository; they're always part of the openwrt repository.

Another advantage of moving this to the feed is that @blogic will review it. And he will (hopefully) make very appropriate comments like: this is the wrong place to set UCI defaults; that's something that should be done by a script in /etc/uci-defaults.

Reply:
as part of handling the bring-up on UGW that adds the support in init.d I will split the script functionality as requested here to:
1. repeater_mode configurations to prplmesh.sh
2. UCI defaults to the right location of UCI configurations (need to define what are defaults and what are according to mode of operation)